WordPress/gutenberg
# Prompt to keep or replace Featured Image when replacing Image block media that is set as the Featured Image

## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/3930
* https://github.com/wordpress-mobile/WordPress-iOS/pull/17132
* https://github.com/wordpress-mobile/WordPress-Android/pull/15303

## Description
Fixes #34429. If an Image block is set as the post's Featured Image, replacing the Image block's media will prompt the user as to whether they want to replace the Featured Image as well. 

## How has this been tested?

⚠️ Part of the changes in this PR focus specifically on images that need to be uploaded to the site (e.g. from a free image service), as they operate differently than images already found within the site's Media Library. So, it is likely a good idea to test a few different paths to replacing the Image block media. 

### Scenarios

- Uploading a new image from device...
- Selecting an existing image within the Media Library... 
- Switching between _the same_ two images within the Media Library multiple times over...
- Editing the current image for the _Image_ block...

### Testing Steps

1. Launch the post editor. 
2. Add an Image block. 
3. Set media for the block (Image A). 
4. Open the block settings. 
5. Tap "Set as Featured Image."
6. ✅ Verify "Featured Image" badge displays atop image. 
6. Tap three-dot menu in top right of editor. 
7. Tap "Post Settings."
8. ✅ Verify the _Image A_ is set as the Featured Image. 
9. Change the media for the block (Image B). N.b. See "Scenarios" above for various ways to perform this action.
10. Tap "Replace with new image." 
11. ✅ Verify "Featured Image" badge displays atop image. 
11. Tap three-dot menu in top right of editor. 
12. Tap "Post Settings."
13. ✅ Verify _Image B_ is set as the Featured Image. 
14. Change the media for the block (Image C). 
15. Tap "Keep current featured image." 
16. ✅ Verify "Featured Image" badge _no longer displays_ atop image. 
16. Tap three-dot menu in top right of editor. 
17. Tap "Post Settings."
18. ✅ Verify _Image B_ remains set as the Featured Image. 

## Screenshots 

| iOS | Android |
| - | - |
| ![iOs replace featured image prompt](https://user-images.githubusercontent.com/438664/151286523-c8dac311-d205-42f1-abda-798a9aef897d.jpeg) | ![Android replace featured image prompt](https://user-images.githubusercontent.com/438664/151284874-d43711e7-ec12-49fd-acdd-efb0f122598e.jpg) |

## Types of changes
New feature

## Checklist:
